pcsx2-qt/GameList/GameListWidget.cpp

```diff
@@ -293,6 +298,137 @@ void GameListWidget::initialize()
+	if (image->width() == dpr_expected_width && image->height() == dpr_expected_height)
+		return;
+
+	// Resize
+	if ((static_cast<float>(image->width()) / static_cast<float>(image->height())) >=
+		(static_cast<float>(dpr_expected_width) / static_cast<float>(dpr_expected_height)))
+	{
+		*image = image->scaledToWidth(dpr_expected_width, Qt::SmoothTransformation);
+	}
+	else
+	{
+		*image = image->scaledToHeight(dpr_expected_height, Qt::SmoothTransformation);
+	}
+
+	if (image->width() == dpr_expected_width && image->height() == dpr_expected_height)
+		return;
+
+	// Padding
+	int xoffs = 0;
+	int yoffs = 0;
+	const int image_width = image->width();
+	const int image_height = image->height();
+	if (image_width < dpr_expected_width)
+		xoffs = static_cast<int>(static_cast<qreal>((dpr_expected_width - image_width) / 2) / dpr);
+	if (image_height < dpr_expected_height)
+		yoffs = static_cast<int>(static_cast<qreal>((dpr_expected_height - image_height) / 2) / dpr);
+
+	QImage padded_image(dpr_expected_width, dpr_expected_height, QImage::Format_ARGB32);
+	padded_image.setDevicePixelRatio(dpr);
+	if (fill_with_top_left)
+		padded_image.fill(image->pixel(0, 0));
+	else
+		padded_image.fill(Qt::transparent);
+
+	// Painting
+	QPainter painter;
+	const float opacity = Host::GetBaseFloatSettingValue("UI", "GameListBackgroundOpacity");
+	if (painter.begin(&padded_image))
+	{
+		painter.setOpacity(opacity);
+		painter.setCompositionMode(QPainter::CompositionMode_Source);
+		painter.drawImage(xoffs, yoffs, *image);
+		painter.end();
+	}
+
+	*image = std::move(padded_image);
+}
+
+void GameListWidget::setCustomBackground()
+{
+	std::string path = Host::GetBaseStringSettingValue("UI", "GameListBackgroundPath");
+	bool enabled = Host::GetBaseBoolSettingValue("UI", "GameListBackgroundEnabled");
+
+	if (!Path::IsAbsolute(path))
+		path = Path::Combine(EmuFolders::DataRoot, path);
+
+	// Cleanup old animation if it still exists on gamelist
+	if (m_background_movie != nullptr)
+	{
+		delete m_background_movie;
+		m_background_movie = nullptr;
+	}
+
+	// Only try to create background if both 1. path and 2. setting are valid
+	if (!path.empty() && enabled)
+	{
+		QMovie* new_movie = new QMovie(QString::fromStdString(path));
+		if (new_movie->isValid())
+		{
+			m_background_movie = new_movie;
+		}
+		else
+		{
+			Console.Warning("Failed to load background movie from: %s", path.c_str());
+			delete new_movie;
+		}
+	}
+
+	// If there is no valid background  then reset fallback to UI state
+	if (!m_background_movie)
+	{
+		m_ui.stack->setPalette(QApplication::palette());
+		m_table_view->setAlternatingRowColors(true);
+		return;
+	}
+
+	// Background is valid, connect the signals and start animation in gamelist
+	connect(m_background_movie, &QMovie::frameChanged, this, [this](int frameNum) { processBackgroundFrames(frameNum); });
+
+	m_background_movie->start();
+	m_table_view->setAlternatingRowColors(false);
+}
+
+void GameListWidget::updateCustomBackgroundState()
```

Comment:
```
void GameListWidget::updateCustomBackgroundState()
{
	if (!m_background_movie)
		return;

	if (isVisible() && QtHost::isFocused())
	{
		m_background_movie->start();
	}
	else
	{
		m_background_movie->stop();
	}
}
```

Might stop the weird behaviour to do that animation only plays if gamelist is visible AND window is focused to avoid race condition.

Might by trying to call isVisible() on the class itself (GameListWidget section) rather than on an instance or even change event.